src/main.ts

```diff
@@ -437,6 +437,7 @@ class NixInstallerAction {
         const fileStream = fs.createWriteStream(tempfile);
         const fileStreamWeb = stream.Writable.toWeb(fileStream);
         await response.body.pipeTo(fileStreamWeb);
+        fileStream.end();
```

Comment:
Referring to https://nodejs.org/api/fs.html#filehandlecreatewritestreamoptions, I'm not sure this syncs the file. I wonder if we also need to call...
https://nodejs.org/api/fs.html#filehandlesync

Reply:
what interesting is that the code I changed wasn't calling sync explicitly, and was just waiting for the finished signal. (which I'd assume is similar to awaiting the `pipeTo`)

I'm going to going to really wack this with a big hammer, open the file so I get a filehandle, make sure it gets fsync'd explicitly and see how this fares.

Reply:
further musing: there's a "synchronous append" mode (`as`) for opening, but not an equivalent for `w`? But there is `rs+`:

> 'rs+': Open file for reading and writing in synchronous mode. Instructs the operating system to bypass the local file system cache.